fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -99,6 +99,9 @@
         this.executor = executor;
         this.limitManager = new CursorLimitManager(state.context, scanProperties);
         this.limitRemaining = scanProperties.getExecuteProperties().getReturnedRowLimitOrMax();
+        if (this.limitRemaining == Integer.MAX_VALUE) {
+            this.limitRemaining = 200;
```

Comment:
That does not seem right. We don't want the whole cursor to stop before the supplied limit. We just want to limit the individual searches and repeat within a single cursor lifetime.